src/app/cli/src/coda_main.ml

```diff
@@ -1123,12 +1129,39 @@ module Run (Config_in : Config_intf) (Program : Main_intf) = struct
         io_field "sync_status" ~typ:sync_status
           ~args:Arg.[]
           ~resolve:(fun {ctx= coda; _} () ->
-            Deferred.Result.return (Program.sync_status coda) )
+            Deferred.return
+              (Inputs.Incr_status.Observer.value @@ Program.sync_status coda)
+            >>| Result.map_error ~f:Error.to_string_hum )
 
       let commands = [sync_state]
     end
 
-    let schema = Graphql_async.Schema.(schema Queries.commands)
+    module Subscriptions = struct
+      let to_pipe observer =
+        let reader, writer = Async.Pipe.create () in
+        Incr_status.Observer.on_update_exn observer ~f:(function
+          | Initialized value ->
+              Async.Pipe.write_without_pushback writer value
```

Comment:
Here we want `write_or_drop_head` rather than write_without_pushback. We don't want buffering we just want to get the head.